ipset/ipset.c

```diff
@@ -170,42 +180,88 @@ static int add_to_ipset(filter_dev dev, const char *setname, const void *ipaddr,
 	mnl_attr_put(nlh, (af == AF_INET ? IPSET_ATTR_IPADDR_IPV4 : IPSET_ATTR_IPADDR_IPV6)
 			| NLA_F_NET_BYTEORDER, (af == AF_INET ? sizeof(struct in_addr) : sizeof(struct in6_addr)), ipaddr);
 	mnl_attr_nest_end(nlh, nested[1]);
+    if (ttl >= 0) {
+        nested[2] = mnl_attr_nest_start(nlh, IPSET_ATTR_TIMEOUT);
+        mnl_attr_put(nlh, NLA_F_NET_BYTEORDER, sizeof(time_t), &ttl);
```

Comment:
**[Note]** Seems I have an old version of these changes here, should just be a single u32 packed attribute. I'll update this when I get a chance later today.